### PR TITLE
[Support] cleanup tests

### PR DESCRIPTION
**Pull request by mirceanis**
### What's here

This represents the first pass at upgrading our test suite to be a bit more expressive and less flaky.
The goals are:

 * [X] move any test that doesn't use `AndroidKeyStore` into JVM tests
 * [X] use better assertion syntax provided by `assertK`
 * [X] replace `mockito` with `mockK` for a more idiomatic and simpler syntax
 * [X] use canned responses instead of calling backends during JVM tests

### Testing

As usual, running `./gradlew test cC` should go through the whole test suite.

### Notes

The tests in `signer` and `demoapp` haven't been touched. They represent a significant portion of the test suite and will be the goal of a new PR.

There is a warning shown by gradle during configuration `WARNING: The option setting 'android.enableUnitTestBinaryResources=true' is experimental and unsupported.`
This has to be ignored until we move to android gradle plugin 3.4 (currently using 3.3 stable)

### CODE REVIEW

File: sdk/src/test/java/me/uport/sdk/AccountStorageTest.kt [hunk 1]
```diff
@@ -45,22 +47,21 @@ class AccountStorageTest {
                 ""
         )
 
-        storage.upsert(account)
+        storage.upsert(refAccount)
+        storage.upsert(refAccount)
```
Comment (ugoamanoh) [inline sdk/src/test/java/me/uport/sdk/AccountStorageTest.kt]: `storage.upsert(refAccount)` is called twice

File: uport-did/src/main/java/me/uport/sdk/uportdid/UportIdentityDocument.kt [hunk 2]
```diff
@@ -113,4 +113,22 @@ data class UportDIDDocument(
         override val context: String = "https://w3id.org/did/v1",
 
         val uportProfile: UportIdentityDocument
-) : DIDDocument
\ No newline at end of file
+) : DIDDocument {
```
Comment (ugoamanoh) [inline uport-did/src/main/java/me/uport/sdk/uportdid/UportIdentityDocument.kt]: We should keep the naming consistent
`UportIdentityDocument` -> `UportDIDDocument`

File: identity/src/androidTest/java/me/uport/sdk/identity/KPAccountCreatorTest.kt [hunk 3]
```diff
@@ -21,12 +27,15 @@ class KPAccountCreatorTest {
...
-            assertTrue(account.type == AccountType.KeyPair)
-            assertTrue(account.address.isNotEmpty())
-            assertTrue(account.publicAddress.isNotEmpty())
-            assertTrue(account.deviceAddress.isNotEmpty())
+
+            assert(account).all {
+                isNotNull()
+                isNotEqualTo(Account.blank)
+            }
+            assert(account.type == AccountType.KeyPair)
```
[the remark this answers is not in the document]
Comment (mirceanis) [reply identity/src/androidTest/java/me/uport/sdk/identity/KPAccountCreatorTest.kt]: right, I missed this one :)

File: sdk/src/test/java/me/uport/sdk/AccountStorageTest.kt (hunk 1 above)
Comment (mirceanis) [reply sdk/src/test/java/me/uport/sdk/AccountStorageTest.kt]: good catch, it used to be called twice before too and I missed it during my refactor

File: uport-did/src/main/java/me/uport/sdk/uportdid/UportIdentityDocument.kt (hunk 2 above)
Comment (mirceanis) [reply uport-did/src/main/java/me/uport/sdk/uportdid/UportIdentityDocument.kt]: I don't understand the inconsistency.
The `UportIdentityDocument` is the legacy model for the profile document, that didn't conform to DID standards.
`UportDIDDocument` conforms to the standards and can be obtained from the legacy model.

I'll move it to a different file and apply some kdoc to clear up misunderstandings

File: https-did/src/main/java/me/uport/sdk/httpsdid/HttpsDIDResolver.kt [hunk 4]
```diff
@@ -1,43 +1,34 @@
...
-                    continuation.resume(HttpsIdentityDocument.fromJson(ddo)!!)
-                }
-            }
+            val ddoString = getProfileDocument(domain)
+            val ddo = HttpsDIDDocument.fromJson(ddoString)
+            return ddo
+                    ?: throw BlankDocumentError("no profile document found for `$did`")
         } else {
-            continuation.resumeWithException(java.lang.IllegalArgumentException("The DID('$did') cannot be resolved by the uPort DID resolver"))
+            throw DidResolverError("The DID('$did') cannot be resolved by the HTTPS DID resolver")
```
[the remark this answers is not in the document]
Comment (mirceanis) [reply https-did/src/main/java/me/uport/sdk/httpsdid/HttpsDIDResolver.kt]: In the error message, `$did` is wrapped in apostrophes and brackets already. I don't think it needs more space

**Pull request merged**